rocks/RocksIterator.java

```diff
@@ -75,40 +76,55 @@ public final boolean hasNext() {
                 return true;
             case EMPTY:
                 return fetchAndCheck();
+            case SEEKED_EMPTY:
+                return checkValidNext();
```

Comment:
after we have done a `seek()`, we want to lazily check whether there is a valid next element without actually calling `next()`. To introduce this we create a new state `SEEKED_EMPTY` which does half the work of `EMPTY`, just checking iterator validity without progressing the rocks iterator to the next element